lib/Frontend/nvqpp/ConvertExpr.cpp

```diff
@@ -1948,6 +1984,45 @@ bool QuakeBridgeVisitor::VisitCXXConstructExpr(clang::CXXConstructExpr *x) {
         assert((isa<quake::RefType, quake::VeqType>(tosTy)));
         return true;
       }
+
+      if (ctorName == "vector") {
+        // This is a std::vector constructor, first we'll check if it
+        // is constructed from a constant initializer list, in that case
+        // we'll have a AllocaOp at the top of the stack that allocates a
+        // ptr<array<TxC>>, where C is constant / known
+        if (auto alloca = dyn_cast<cc::AllocaOp>(peekValue().getDefiningOp())) {
+          auto asPtrType =
+              dyn_cast<cc::PointerType>(alloca.getAddress().getType());
+          if (auto arrayTy =
+                  dyn_cast<cc::ArrayType>(asPtrType.getElementType()))
+            if (alloca.getNumOperands() > 0)
+              return pushValue(builder.create<cc::StdvecInitOp>(
+                  loc, cc::StdvecType::get(arrayTy.getElementType()), alloca,
+                  alloca.getSeqSize()));
+        }
+
+        // Next check if its created from a size integer
+        if (peekValue().getType().isIntOrFloat()) {
```

Comment:
And here, similarly. Make sure this is the `std::vector<T>::vector(size_t)` ctor...

Reply:
See latest.